src/Sylius/Component/Core/Generator/ImagePathGeneratorInterface.php

```diff
@@ -0,0 +1,21 @@
+<?php
+
+/*
+ * This file is part of the Sylius package.
+ *
+ * (c) Paweł Jędrzejewski
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+declare(strict_types=1);
+
+namespace Sylius\Component\Core\Generator;
+
+use Sylius\Component\Core\Model\ImageInterface;
+
+interface ImagePathGeneratorInterface
+{
+    public function generate(ImageInterface $image): string;
```

Comment:
In `ImageUploader` we use this method in a way that needs to be defined further, possibly as a comment.

This method cannot be idempotent, every call to this method should return a different value. If the same value is returned, then `ImageUploader` might enter an infinite loop.